client/elasticClient.go

```diff
@@ -323,6 +329,45 @@ func (ec *elasticClient) createAlias(alias string, index string) error {
 	return parseResponse(res, nil, elasticDefaultErrorResponseHandler)
 }
 
+func (ec *elasticClient) getWriteIndex(alias string) (string, error) {
+	res, err := ec.client.Indices.GetAlias(
+		ec.client.Indices.GetAlias.WithIndex(alias),
+	)
+	if err != nil {
+		return "", err
+	}
+
+	var indexData map[string]struct {
+		Aliases map[string]struct {
+			IsWriteIndex bool `json:"is_write_index"`
+		} `json:"aliases"`
+	}
+
+	err = parseResponse(res, &indexData, elasticDefaultErrorResponseHandler)
+	if err != nil {
+		return "", err
+	}
+
+	// Iterate over the map and find the write index
+	var writeIndex string
+	for index, details := range indexData {
+		if len(indexData) == 1 {
+			return index, nil
+		}
+
+		for _, indexAlias := range details.Aliases {
+			if indexAlias.IsWriteIndex {
+				writeIndex = index
+				break
+			}
+		}
+		if writeIndex != "" {
+			break
+		}
+	}
+	return writeIndex, nil
```

Comment:
with the above suggestions, perhaps return some error here, meaning the index was not found

Reply:
refactored